app/models/journaled/bulk_delivery.rb

```diff
@@ -41,17 +41,95 @@ def kinesis_client
     Journaled::KinesisClient.generate
   end
 
-  def requeue_failed_records!(response) # rubocop:disable Metrics/AbcSize
-    records_with_responses = records.zip(response.records)
-    errored_records = records_with_responses.select { |_record, resp| resp.error_code.present? }.map(&:first)
+  def handle_failures!(response)
+    ErrorHandler.new(response, records, app_name).handle!
+  end
+
+  class ErrorHandler
+    attr_reader :response, :records, :app_name
+
+    def initialize(response, records, app_name)
+      @response = response
+      @records = records
+      @app_name = app_name
+    end
```

Comment:
Is there ever a chance that one of these would be left nil? If so, should we make these named?

Reply:
`app_name` may be nil, but will always be supplied if that makes sense.

`app_name` is passed into `bulk_delivery` but maybe passed in as `nil` which we want to carry through


I think using named args is def a good idea here in general cause 3 params probably want to be named. BUTT I might leave this for now, just cause this is an internal class that isn't supposed to be consumed anywhere else than this single usage. But if we were to ever extract this into a 'full' class I :100:% agree that we should make this keyword args!

Reply:
Alrighty! Sounds reasonable :)